datatree/datatree.py

```diff
@@ -1185,6 +1190,51 @@ def filter(self: DataTree, filterfunc: Callable[[DataTree], bool]) -> DataTree:
         }
         return DataTree.from_dict(filtered_nodes, name=self.root.name)
 
+    def match(self, pattern: str) -> DataTree:
+        """
+        Return nodes with paths matching pattern.
+
+        Uses unix glob-like syntax for pattern-matching.
+
+        Parameters
+        ----------
+        pattern: str
+            A pattern to match each node path against.
+
+        Returns
+        -------
+        DataTree
+
+        See Also
+        --------
+        filter
+        pipe
+        map_over_subtree
+
+        Examples
+        --------
+        >>> dt = DataTree.from_dict(
+        >>>     {
```

Comment:
@keewis blackdoc raised an error here:

```
error: cannot format /code/datatree/datatree.py: Cannot parse: 1217:0: EOF in multi-line statement
```

Reply:
Oh - I copied this from syntax used in the [docstring of xarray's `Dataset.map`](https://github.com/pydata/xarray/blob/ccc8f9987b553809fb6a40c52fa1a8a8095c8c5f/xarray/core/dataset.py#L6842). So I guess it should look like this [example from pint-xarray](https://github.com/xarray-contrib/pint-xarray/blob/963e0fc7eae7c90257040ecd7756c606f55aa2ac/pint_xarray/accessors.py#L292). Thanks!

Reply:
Actually no I'm just being an idiot